aioamqp/protocol.py

```diff
@@ -317,9 +321,13 @@ def channel(self, **kwargs):
         """Factory to create a new channel
 
         """
-        if not len(self.channels_ids):
-            raise exceptions.NoChannelAvailable()
-        channel_id = self.channels_ids.pop(0)
+        try:
+            channel_id = self.channels_ids_free.pop()
+        except KeyError:
+            if self.channels_ids_ceil >= amqp_constants.MAX_CHANNELS:
```

Comment:
I've read some more documentation on rabbitmq:

https://www.rabbitmq.com/amqp-0-8-to-0-9-1.html
"The meaning of "channel-max" has been clarified: for example, if channel-max is set to 2, then it is illegal to use channels numbered 3 or higher for frames carried by the connection. "

To clarify, 0 is a special channel linked to the connection. So the code is correct.

Moreover, I'll create a new issue, because we can configure the connection to allow for more channel, using tune / tune-ok commands. We're using a constant but it's configuratble:

https://www.rabbitmq.com/amqp-0-9-1-reference.html#connection.tune
https://www.rabbitmq.com/amqp-0-9-1-reference.html#connection.tune-ok